service/src/intTest/java/uk/nhs/adaptors/gp2gp/gpc/GpcFindDocumentsComponentTest.java

```diff
@@ -0,0 +1,172 @@
+@SpringBootTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+public class GpcFindDocumentsComponentTest extends BaseTaskTest {
+    private static final String NHS_NUMBER_WITH_DOCUMENT = "9690937286";
+    private static final String NHS_NUMBER_WITHOUT_DOCUMENT = "9690937294";
+    private static final String NHS_NUMBER_INVALID = "ASDF";
+    private static final String PATIENT_NOT_FOUND = "9876543210";
+    private static final String VALID_DOCUMENT_ID = "07a6483f-732b-461e-86b6-edb665c45510";
+    private static final String INVALID_NHS_NUMBER = "INVALID_NHS_NUMBER";
+
+    @Autowired
+    private GetGpcDocumentReferencesTaskExecutor gpcFindDocumentsTaskExecutor;
+    @Autowired
+    private EhrExtractStatusRepository ehrExtractStatusRepository;
+    @Autowired
+    private GpcConfiguration configuration;
+
+    private EhrExtractStatus setupDatabase() {
+        var ehrExtractStatus = EhrExtractStatusTestUtils.prepareEhrExtractStatus();
+        ehrExtractStatus.setGpcAccessDocument(EhrExtractStatus.GpcAccessDocument.builder()
+            .documents(List.of(
+                EhrExtractStatus.GpcAccessDocument.GpcDocument.builder()
+                    .documentId(EhrStatusConstants.DOCUMENT_ID)
+                    .accessDocumentUrl(EhrStatusConstants.GPC_ACCESS_DOCUMENT_URL)
+                    .build()
+            ))
+            .build());
+        return ehrExtractStatusRepository.save(ehrExtractStatus);
+    }
+
+    @Test
+    public void When_FindDocumentTaskIsStartedForPatientWithDocument_Expect_DatabaseToBeUpdated() {
+        var ehrExtractStatus = setupDatabase();
+        assertThatAccessRecordWasOverwritten(ehrExtractStatus);
+        var taskDefinition = buildFindDocumentTask(ehrExtractStatus, NHS_NUMBER_WITH_DOCUMENT);
+        gpcFindDocumentsTaskExecutor.execute(taskDefinition);
+
+        var updatedEhrExtractStatus = ehrExtractStatusRepository.findByConversationId(taskDefinition.getConversationId()).get();
+        assertThatAccessRecordWasUpdated(updatedEhrExtractStatus, ehrExtractStatus, taskDefinition);
+        Mockito.verify(taskDispatcher).createTask(buildGetDocumentTask(taskDefinition));
+
+    }
+
+    @Test
+    public void When_FindDocumentTaskIsStartedForPatientWithoutDocument_Expect_DatabaseToNotBeUpdated() {
+        var ehrExtractStatus = setupDatabase();
+        assertThatAccessRecordWasOverwritten(ehrExtractStatus);
+        var taskDefinition = buildFindDocumentTask(ehrExtractStatus, NHS_NUMBER_WITHOUT_DOCUMENT);
+        gpcFindDocumentsTaskExecutor.execute(taskDefinition);
+
+        var updatedEhrExtractStatus = ehrExtractStatusRepository.findByConversationId(taskDefinition.getConversationId()).get();
+        assertThat(updatedEhrExtractStatus.getGpcAccessDocument().getDocuments().size()).isEqualTo(0);
+
+    }
+
+    @Test
+    public void When_FindDocumentTaskIsStarted_Expect_PatientNotFound() {
+        var ehrExtractStatus = setupDatabase();
```

Comment:
I do not like the name of this method. We are not technically setting up a database, but adding test data to it. How about "addTestDataToDatabase()"?